loggers/pino/test/apm.test.js

```diff
@@ -323,64 +323,30 @@ test('apmIntegration=false disables tracing integration', t => {
   })
 })
 
-test('can override service.name, event.dataset', t => {
```

Comment:
Yah, good find. After learning more about the OTel Logs SDK expected usage, and learning about the ecs-logging work for **Java** -- which was done by Felix who, IIUC, created the ecs-logging idea -- I believe that I misinterpreted this from the ecs-logging spec for `service.name` and others (https://github.com/elastic/ecs-logging/blob/main/spec/spec.json#L75-L83):

```
            "comment": [
                "Configurable by users.",
                "When an APM agent is active, it should auto-configure this field if not already set."
            ]
```

My old understanding was that a user could do `log.info({service:{name:"myservice"}}, "some message")` and our formatter would go to lengths to ensure we didn't conflict with that given value.

My new understanding is that "Configurable by users" is when configuring the *formatter*, i.e. when doing:

```
var log = pino(ecsFormat(/* ... these config options ... */));
```

My intent is to add `serviceName`, `serviceVersion`, etc. config options to the `ecsFormat()` functions, as you see in the Java ecs-logging docs here: https://www.elastic.co/guide/en/ecs-logging/java/current/setup.html#_use_the_ecs_encoderformatterlayout

This change -- not guarding against `service.name` collisions in every log record -- simplifies and speeds up these formatters a lot.

Reply:
> This change -- not guarding against `service.name` collisions in every log record -- simplifies and speeds up these formatters a lot.

The downside is that a user doing `log.info({service:{name:"myservice"}}, "some message")` and with APM integration will now get a log record with `,"service":{"name":"myservice"},`  **and** `"service.name":"...the serviceName configured in APM ..."`. This could cause some confusion.  However, it gets us to the same behaviour as the ecs-logging-java libs.   Also it feels like it will be equivalent to what future users of the OTel Logs API/SDK will get -- where resource attributes like "service name" are configured on the Logger Provider, and not on each `log.info(...)` call.